extension/modules/personas.js

```diff
@@ -0,0 +1,1480 @@
+ * of those above. If you wish to allow use of your version of this file only
+ * under the terms of either the GPL or the LGPL, and not to allow others to
+ * use your version of this file under the terms of the MPL, indicate your
+ * decision by deleting the provisions above and replace them with the notice
+ * and other provisions required by the GPL or the LGPL. If you do not delete
+ * the provisions above, a recipient may use your version of this file under
+ * the terms of any one of the MPL, the GPL or the LGPL.
+ *
+ * ***** END LICENSE BLOCK ***** */
+ 
+ var EXPORTED_SYMBOLS = ["PersonaController"];
+
+// Generic modules get imported into the persona controller rather than
+// the global namespace after the controller definition below so they don't
+// conflict with modules with the same names imported by other extensions.
+
+// Define Components as var here as they are already defined for Firefox. See Bug 484062 for details.
+if (typeof Cc == "undefined")
+    var Cc = Components.classes;
+if (typeof Ci == "undefined")
+    var Ci = Components.interfaces;
+if (typeof Cr == "undefined")
+    var Cr = Components.results;
+if (typeof Cu == "undefined")
+    var Cu = Components.utils;
+
+Components.utils.import("resource://gre/modules/Services.jsm");
+try{var console = (Cu.import("resource://gre/modules/Console.jsm", {})).console;}
+catch(error){var console = (Cu.import("resource://gre/modules/"+"devtools"+"/Console.jsm", {})).console;}
+
+// It's OK to import the service module into the global namespace because its
+// exported symbols all contain the word "persona" (f.e. PersonaService).
+Cu.import("resource://personas/modules/service.js");
+	
+var PersonaController = {
+    _previewTimeoutID: null,
+    _resetTimeoutID: null,
+
+    //**************************************************************************//
+    // Shortcuts
+
+    // Generic modules get imported into these properties rather than
+    // the global namespace so they don't conflict with modules with the same
+    // names imported by other extensions.
+    Observers: null,
+    Preferences: null,
+    StringBundle: null,
+    URI: null,
+    LightweightThemeManager: null,
+
+    // Access to extensions.personas.* preferences.  To access other preferences,
+    // call the Preferences module directly.
+    get _prefs() {
+        delete this._prefs;
+        return this._prefs = new this.Preferences("extensions.personas.");
+    },
+
+    get _strings() {
+        delete this._strings;
+        return this._strings = new this.StringBundle("chrome://personas/locale/personas.properties");
+    },
+
+    get _brandStrings() {
+        delete this._brandStrings;
+        return this._brandStrings =
+            new this.StringBundle("chrome://branding/locale/brand.properties");
+    },
+
+    get _menu() {
+        delete this._menu;
+        return this._menu = document.getElementById("personas-menu");
+    },
+    get_menu(document) {
```

Comment:
Again, please *stop* doing that. If you rewrite code, delete the old one.
Diff-tools are supposed to handle diffs, and they are actually better at it
than humans.

On Mon, May 30, 2016 at 2:15 PM, derinb <notifications@github.com> wrote:

> In extension/modules/personas.js
> <https://github.com/mozilla/personas-plus/pull/24#discussion_r65063839>:
>
> > +    get _strings() {
> > +        delete this._strings;
> > +        return this._strings = new this.StringBundle("chrome://personas/locale/personas.properties");
> > +    },
> > +
> > +    get _brandStrings() {
> > +        delete this._brandStrings;
> > +        return this._brandStrings =
> > +            new this.StringBundle("chrome://branding/locale/brand.properties");
> > +    },
> > +
> > +    get _menu() {
> > +        delete this._menu;
> > +        return this._menu = document.getElementById("personas-menu");
> > +    },
> > +    get_menu(document) {
>
> But those are getter functions. Getter functions does not accept
> parameters.
>
> I just leaved getter functions so that you can compare the difference.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24/files/0e46a799888a86b66555e538330edc80a3fc6aac#r65063839>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKmgyVXZpaiAujiyeU_fRsb6bRgWgks5qGtTzgaJpZM4Iout8>
> .
>